### PR TITLE
Merge arduinos

### PR DESCRIPTION
**Pull request by maurin**
This branch represents the refactor necessary to combine all of the surface and rover code in to a single shared code base.  Additionally, it relies on the use of the arduino-cli tool to compile.

### CODE REVIEW

File: arduino/arduino.ino [hunk 1]
```diff
@@ -1,52 +1,43 @@
...
 {
 	// Initialize main subsystems
 	NArduinoManager::Initialize();
 	NCommManager::Initialize();
+	NVehicleManager::Initialize();
 	NModuleManager::Initialize();
+	NDataManager::Initialize();
 
+	// Set timer 5 divisor to 8 for PWM frequency of 3921.16Hz (D44, D45, D46)
+	TCCR5B = ( TCCR5B & B11111000 ) | B00000010;
```
Comment (maurin) [inline arduino/arduino.ino:26]: @maurin This linefails to compile when working on an uno.  We will use an UNO for the surface.

Suggestions.
1. remove line if timer are not used in rover and arduino
2. conditionally compile this line depending on target (platform )
3. Move line to initialize code of the class that needs it.

**Review by joshnarayan — COMMENTED** [review]
Hey Pablo, check to see if the modules are actually being loaded - somebody.

File: arduino/SysConfig_Rover.h [hunk 2]
```diff
@@ -47,3 +40,9 @@
 
 // BNO055 IMU
 #define HAS_BNO055 (0)
+
+// Receiver
+#define HAS_RECEIVER (0)
+
+//Some_module
+#define HAS_SOME_MODULE (0)
```
Comment (CaitlynFong) [inline arduino/SysConfig_Rover.h:48]: Double check that the correct modules are being enabled and disabled.

File: arduino/BuildErrorMessage.h [hunk 3]
```diff
@@ -0,0 +1,6 @@
+/*
+This will always generate a compile time error message from the pre-processor.
+It is okay to include more than once.  Hence, no include guards
+*/
+
+#error No build flag defined
```
Comment (maurin) [inline arduino/BuildErrorMessage.h]: @maurin  "Must specify BUILD_ROVER or BUILD_SURFACE at compile time"
Update the error to make it more user friendly

File: arduino/ThrusterConfigDefs.h [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+The following define the values for each possible configuration.
+When a new CThruster_XXX module is created, it will be necessary
+to add a new definition here.
+*/
+
+#define THRUSTER_CONFIG_NONE    0
+#define THRUSTER_CONFIG_2X1     1
+#define THRUSTER_CONFIG_2Xv2    2
+#define THRUSTER_CONFIG_v2X1Xv2 3
```
Comment (maurin) [inline arduino/ThrusterConfigDefs.h]: Protect each value with parens

File: arduino/arduino.ino [hunk 5]
```diff
@@ -1,52 +1,43 @@
...
-
 {
 	// Reset the watchdog timer
 	wdt_reset();
 
 	// Attempt to read a current command off of the command line
 	NCommManager::GetCurrentCommand();
 
+	// Handle any config change requests
+	NVehicleManager::HandleMessages( NCommManager::m_currentCommand );
```
Comment (maurin) [inline arduino/arduino.ino:42]: Add a comment about possibly removing this line.  
Make this  a "TODO" comment

File: arduino/SysConfig_Rover.h (hunk 2 above)
Comment (maurin) [reply arduino/SysConfig_Rover.h:48]: Absolutely we will.  For now, let's keep all the modules turned off.  We'll turn them on as we build out the surface and rover functionality.

**Review by joshnarayan — COMMENTED** [review]
I'm not sure why but this is really bad

**Review by GarrettWeber — APPROVED** [review]
This is great! Will make it easier to work with this code in the future.

**Pull request merged**